code/game/objects/items/rcd/RPD.dm

```diff
@@ -273,15 +298,114 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 /obj/item/pipe_dispenser/attack_self(mob/user)
 	ui_interact(user)
 
-/obj/item/pipe_dispenser/pre_attack_secondary(obj/machinery/atmospherics/target, mob/user, params)
+/obj/item/pipe_dispenser/AltClick(mob/user)
+	. = ..()
+	if(!(upgrade_flags & RPD_UPGRADE_SPEEDPIPE))
+		return .
+	toggle_speedpipe_marking(user)
+
+/obj/item/pipe_dispenser/alt_click_secondary(mob/user)
+	. = ..()
+	speedpipe_build(user)
+
+/obj/item/pipe_dispenser/CtrlClick(mob/user)
+	. = ..()
+	clear_out_speedpipe(user)
```

Comment:
why are we returning the default value if we are performing actions based on the click. We do not want the attack chain to continue here